src/components/date/date.pw.tsx

```diff
@@ -0,0 +1,1349 @@
+      await inputParent.click();
+      const arrowElement = getDataElementByValue(page, arrow);
+      await arrowElement.click();
+      const pickerHeading = page.locator(".DayPicker-Caption").locator("div");
+      if (month === "next") {
+        await expect(pickerHeading).toHaveText(NEXT_MONTH);
+      } else if (month === "previous") {
+        await expect(pickerHeading).toHaveText(PREVIOUS_MONTH);
+      } else {
+        throw new Error("Only Next or Previous month can be applied");
+      }
+    });
+  });
+
+  arrowKeys.forEach((key) => {
+    test(`should not change the displayed month when ${key} is pressed and next button is focused`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DateInputCustom />);
+
+      const inputParent = getDataElementByValue(page, "input").locator("..");
+      await inputParent.click();
+      const rightArrow = getDataElementByValue(page, "chevron_right").locator(
+        ".."
+      );
+      await rightArrow.focus();
+      await rightArrow.press(key);
+      const pickerHeading = page.locator(".DayPicker-Caption").locator("div");
+      await expect(pickerHeading).toHaveText("May 2022");
+    });
+  });
+
+  arrowKeys.forEach((key) => {
+    test(`should not change the displayed month when ${key} is pressed and previous button is focused`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DateInputCustom />);
+
+      const inputParent = getDataElementByValue(page, "input").locator("..");
+      await inputParent.click();
+      const rightArrow = getDataElementByValue(page, "chevron_left").locator(
+        ".."
+      );
+      await rightArrow.focus();
+      await rightArrow.press(key);
+      const pickerHeading = page.locator(".DayPicker-Caption").locator("div");
+      await expect(pickerHeading).toHaveText("May 2022");
+    });
+  });
+
+  test(`should allow a user to tab into the picker and through its controls`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputCustom value="12/12/2022" />);
+
+    await page.focus("body");
+    await page.keyboard.press("Tab");
+    const input = getDataElementByValue(page, "input");
+    await expect(input).toBeFocused();
+    await page.keyboard.press("Tab");
+    const arrowLeft = getDataElementByValue(page, "chevron_left").locator("..");
+    await expect(arrowLeft).toBeFocused();
+    await page.keyboard.press("Tab");
+    const arrowRight = getDataElementByValue(page, "chevron_right").locator(
+      ".."
+    );
+    await expect(arrowRight).toBeFocused();
+    await page.keyboard.press("Tab");
+    const dayPicker = page.locator(".DayPicker-Day--selected");
+    await expect(dayPicker).toBeFocused();
+  });
+
+  test(`should close the picker and focus the next element in the DOM when focus is on a day element and tab pressed`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputWithButton />);
+
+    await page.focus("body");
+    await page.keyboard.press("Tab");
```

Comment:
**suggestion**: the Cypress version of this test checked which elements were focused at each of these tab stops. I'd prefer we did that in Playwright too, not just to check that the picker is tabbed through as expected but because the way you have it it's not obvious that the picker was ever opened, in which case the assertion that it's closed after all those tabs isn't really very useful. (This test could still pass even if we broke everything so that the picker never opened.)